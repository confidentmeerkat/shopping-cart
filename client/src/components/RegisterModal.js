import React from 'react';
import Modal from 'react-modal';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import '../styles/RegisterModal.css';

const RegisterModal = (props) => (
  <Modal
    className="register-modal"
    isOpen={props.isOpen}
    onRequestClose={props.onRequestClose}
  >
    <form className="form" action="/auth/register" method="POST">
      <h1>Register</h1>
      <TextField
        hintText="Enter Username"
        floatingLabelText="Username"
        name="username"
        autoFocus
      /><br />
      <TextField
        hintText="Enter Password"
        floatingLabelText="Password"
        name="password"
        type="password"
      /><br />
      <TextField
        hintText="Enter E-mail"
        floatingLabelText="E-mail"
        name="email"
      /><br />
      <TextField
        hintText="Enter Address"
        floatingLabelText="Address"
        name="address"
      /><br />
      <TextField
        hintText="Enter Telephone Number"
        floatingLabelText="Telephone Number"
        name="phone"
      /><br />
      <RaisedButton 
        className="btn" 
        label="Submit" 
        primary={true} 
        type="submit"
      />
      <p>Already have an account? <a onClick={props.switch}>Login here</a>.</p>
    </form>
  </Modal>
);

export default RegisterModal;