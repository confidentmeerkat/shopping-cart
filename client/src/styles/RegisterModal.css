.register-modal {
  width: 385px;
  height: 600px;
  margin: auto;
  margin-top: 100px;
  background: white;
  border: 1px solid white;
  border-radius: 5px;
  outline: none;
}

.register-modal .form {
  width: 70%;
  margin: auto;
  margin-top: 10px;
  text-align: center;
  font-family: 'Roboto';
}

.register-modal .form h1 {
  color: #00BCD4;
}

.register-modal .form .btn {
  margin: 30px auto 50px auto;
}

.register-modal .form p {
  margin: 0;
  font-style: italic;
}

.register-modal .form a {
  color: #00BCD4;
  cursor: pointer;
}