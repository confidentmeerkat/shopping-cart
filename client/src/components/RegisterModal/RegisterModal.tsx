import * as React from 'react';
import * as Modal from 'react-modal';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import { ModalProps } from '@typings/modal';
import '@styles/RegisterModal.css';



const RegisterModal = ({ isOpen, onRequestClose, toggle }: ModalProps) => (
  <Modal
    className="register-modal"
    isOpen={isOpen}
    onRequestClose={onRequestClose}
  >
    <form className="form" action="/auth/register" method="POST">
      <h1>Register</h1>
      <TextField
        hintText="Enter Username"
        floatingLabelText="Username"
        name="username"
        autoFocus
      /><br />
      <TextField
        hintText="Enter Password"
        floatingLabelText="Password"
        name="password"
        type="password"
      /><br />
      <TextField
        hintText="Enter E-mail"
        floatingLabelText="E-mail"
        name="email"
      /><br />
      <TextField
        hintText="Enter Address"
        floatingLabelText="Address"
        name="address"
      /><br />
      <TextField
        hintText="Enter Telephone Number"
        floatingLabelText="Telephone Number"
        name="phone"
      /><br />
      <RaisedButton 
        className="btn" 
        label="Submit"
        primary={true} 
        type="submit"
      />
      <p>Already have an account? <a onClick={() => toggle}>Login here</a>.</p>
    </form>
  </Modal>
);

export default RegisterModal;
