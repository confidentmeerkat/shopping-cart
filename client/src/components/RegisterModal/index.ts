import RegisterModal from './RegisterModal';

export default RegisterModal;
