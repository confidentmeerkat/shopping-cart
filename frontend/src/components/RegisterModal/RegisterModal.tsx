import React, { useState } from 'react';
import Modal from 'react-modal';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import { ModalProps } from '@typings/modal';
import { register } from '@api/auth';
import '@styles/RegisterModal.css';

const RegisterModal = ({ isOpen, onRequestClose, setActiveModal, setUser }: ModalProps) => {
  const [data, setData] = useState({});

  const setFormField = (key: string, value: any) => {
    setData({
      ...data,
      [key]: value,
    });
  }

  const onSubmit = () => {
    register(data).then((res) => {
      setUser && setUser(res.data);
      onRequestClose();
    });
  }

  return (
    <Modal
      className="register-modal"
      isOpen={isOpen}
      onRequestClose={onRequestClose}
    >
      <div className="form">
        <h1>Register</h1>
        <TextField
          hintText="Enter Username"
          floatingLabelText="Username"
          autoFocus
          onChange={({ target }: any) => setFormField('username', target.value)}
        /><br />
        <TextField
          hintText="Enter Password"
          floatingLabelText="Password"
          type="password"
          onChange={({ target }: any) => setFormField('password', target.value)}
        /><br />
        <TextField
          hintText="Enter E-mail"
          floatingLabelText="E-mail"
          type="email"
          onChange={({ target }: any) => setFormField('email', target.value)}
        /><br />
        <TextField
          hintText="Enter Address"
          floatingLabelText="Address"
          onChange={({ target }: any) => setFormField('address', target.value)}
        /><br />
        <TextField
          hintText="Enter Telephone Number"
          floatingLabelText="Telephone Number"
          onChange={({ target }: any) => setFormField('phone', target.value)}
        /><br />
        <RaisedButton
          className="btn"
          label="Submit"
          primary={true}
          onClick={onSubmit}
        />
        <p>Already have an account? <a onClick={() => setActiveModal('login')}>Login here</a>.</p>
      </div>
    </Modal>
  );
}

export default RegisterModal;
