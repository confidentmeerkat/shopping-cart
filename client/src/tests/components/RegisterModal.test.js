import React from 'react';
import { shallow } from 'enzyme';
import RegisterModal from '../../components/RegisterModal';

describe('<RegisterModal />', () => {
  it('renders RegisterModal correctly', () => {
    const wrapper = shallow(<RegisterModal />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('form').children()).toHaveLength(13);
  });
});