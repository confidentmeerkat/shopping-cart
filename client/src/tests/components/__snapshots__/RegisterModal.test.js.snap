// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterModal /> renders RegisterModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="register-modal"
  closeTimeoutMS={0}
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <form
    action="/auth/register"
    className="form"
    method="POST"
  >
    <h1>
      Register
    </h1>
    <TextField
      autoFocus={true}
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Username"
      fullWidth={false}
      hintText="Enter Username"
      multiLine={false}
      name="username"
      rows={1}
      type="text"
      underlineShow={true}
    />
    <br />
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Password"
      fullWidth={false}
      hintText="Enter Password"
      multiLine={false}
      name="password"
      rows={1}
      type="password"
      underlineShow={true}
    />
    <br />
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="E-mail"
      fullWidth={false}
      hintText="Enter E-mail"
      multiLine={false}
      name="email"
      rows={1}
      type="text"
      underlineShow={true}
    />
    <br />
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Address"
      fullWidth={false}
      hintText="Enter Address"
      multiLine={false}
      name="address"
      rows={1}
      type="text"
      underlineShow={true}
    />
    <br />
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Telephone Number"
      fullWidth={false}
      hintText="Enter Telephone Number"
      multiLine={false}
      name="phone"
      rows={1}
      type="text"
      underlineShow={true}
    />
    <br />
    <RaisedButton
      className="btn"
      disabled={false}
      fullWidth={false}
      label="Submit"
      labelPosition="after"
      primary={true}
      secondary={false}
      type="submit"
    />
    <p>
      Already have an account? 
      <a>
        Login here
      </a>
      .
    </p>
  </form>
</Modal>
`;
